src/api/users/src/models/celebrateSchema.js

```diff
@@ -1,28 +1,29 @@
 const { celebrate, Joi, Segments } = require('celebrate');
 
+// Generate a display name if none given
+const generateDisplayName = (parent) => `${parent.firstName} ${parent.lastName}`;
+
 const validateId = () =>
   celebrate({
     [Segments.PARAMS]: {
-      id: Joi.string().required(),
+      id: Joi.string().hex().required(),
     },
   });
 
 const validateUser = () =>
   celebrate({
     [Segments.BODY]: Joi.object().keys({
-      id: Joi.number().required(),
       firstName: Joi.string().required(),
       lastName: Joi.string().required(),
-      displayName: Joi.string().default(''),
+      email: Joi.string().email().required(),
+      displayName: Joi.string().default(generateDisplayName),
       isAdmin: Joi.boolean().default(false),
       isFlagged: Joi.boolean().default(false),
-      feeds: Joi.array().items(Joi.string()).required(),
+      feeds: Joi.array().items(Joi.string().uri()).required(),
       github: Joi.object({
         username: Joi.string(),
-        avatarUrl: Joi.string(),
-      }).default({ username: '', avatarUrl: '' }),
```

Comment:
How will having `github.avatarUrl = ''` work better when you do `<img src={github.avatarUrl} />` or `github.username = ''` when you do `fetch('https://api.github.com/user/${github.username})`?  The empty value will belie the fact that we don't have any data.  I would prefer that we not even add `.github` to the user at all if we don't have the data, since it makes checking for it a lot easier: `if(user.github) { /* use the github info */ }`

Reply:
Ah gotcha, makes sense.